src/runner_manager.py

```diff
@@ -281,16 +281,26 @@ def _create_runner(
             except IssueMetricEventError:
                 logger.exception("Failed to issue RunnerInstalled metric")
 
-            fs = shared_fs.get(runner.config.name)
             try:
-                (fs.path / RUNNER_INSTALLED_TS_FILE_NAME).write_text(
-                    str(ts_after), encoding="utf-8"
-                )
-            except FileNotFoundError:
+                fs = shared_fs.get(runner.config.name)
+            except errors.GetSharedFilesystemError:
                 logger.exception(
-                    "Failed to write runner-installed.timestamp, "
-                    "will not be able to issue all metrics."
+                    "Failed to get shared filesystem for runner %s, "
+                    "will not be able to issue all metrics.",
+                    runner.config.name,
                 )
+            else:
+                try:
+                    (fs.path / RUNNER_INSTALLED_TS_FILE_NAME).write_text(
+                        str(ts_after), encoding="utf-8"
+                    )
+                except FileNotFoundError:
+                    logger.exception(
+                        "Failed to write runner-installed.timestamp into shared filesystem "
+                        "for runner %s, will not be able to issue all metrics.",
+                        runner.config.name,
+                    )
```

Comment:
Is there anything we can do to recover in these cases? Like re-create the file system or something?

Reply:
We could try to recreate the fs, but we cannot reattach a disk to a running lxd instance, so we would have to stop and restart the lxd instance and re-register the runner. This also means increasing the complexity of the code, which I am not sure is worth it. Also, if creating the shared fs failed the first time, it is likely that a second attempt will fail as well.
Before this PR, catching the exception was not implemented at all, so we have not seen this issue in production yet.